### PR TITLE
PoC: Show changelog informations

### PR DESCRIPTION
**Pull request by jmleroux**
This is a very early suggestion I made after several questions about fixes and versions.

It is not easy to know wich fix is included in your PIM.
- because you don't know if a fix is part of a tag
- because you don't know if a fix was pulled up

This PR adds a Changelog screen where all changelogs files are merged and ordered by date. This way, you will just have to check in this screen to know if a fix is included in your application

The style is very basic and will be discussed with designers.

Peek 24-11-2019 22-18

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/AnalyticsBundle/Resources/views/changelog/index.html.twig [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{% import 'PimUIBundle:Default:page_elements.html.twig' as elements %}
```
Comment (juliensnz) [inline src/Akeneo/Platform/Bundle/AnalyticsBundle/Resources/views/changelog/index.html.twig]: If we follow the technical guidance from @BitOne I'm not sure we are able to add new screens using twig rendering :/

File: src/Akeneo/Tool/Component/Analytics/Changelog.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Akeneo\Tool\Component\Analytics;
+
+class Changelog
```
Comment (juliensnz) [inline src/Akeneo/Tool/Component/Analytics/Changelog.php]: why not simply using a markdown parser?

Comment (jmleroux) [reply src/Akeneo/Tool/Component/Analytics/Changelog.php]: I thought about it. Why not. It's an implementation detail and this PoC was more to validate the principle of parsing the changelog files and merge them in a new screen.

File: src/Akeneo/Platform/Bundle/AnalyticsBundle/Resources/views/changelog/index.html.twig (hunk 1 above)
Comment (jmleroux) [reply src/Akeneo/Platform/Bundle/AnalyticsBundle/Resources/views/changelog/index.html.twig]: I knew you would say that :laughing:

**Comment by jmleroux** [comment]
I renamed the PR to reflect the proof of concept status

File: src/Akeneo/Tool/Component/Analytics/Changelog.php (hunk 2 above)
Comment (jmleroux) [reply src/Akeneo/Tool/Component/Analytics/Changelog.php]: Would you have one to suggest?

**Comment by SamirBoulil** [comment]
Nice! I personally like the idea (I don't know if it should take this format though) but anyhow this feature should be reviewed by a PM to land in the product.

**Comment by BitOne** [comment]
I really liked the idea, thanks JM.

PIM input would be required, but that's definiitely something that makes sense ;)

**Comment by jjanvier** [comment]
I like the idea :+1: 

But don't forget that our changelogs are almost all "false". For instance, when we fix something in 2.3.54. We don't put the same message on the 3.x during the pullup. We need an automatic changelog. (But it's not related to your PR)

**Comment by ahocquard** [comment]
Can't we have an akeneo page instead, listing all the fixes? Am I the only thinking it's weird to embed it in the application itself :smile: ? Actually, I don't know existing application doing that. Generally you go on a website page with all the changelogs.

For example:
https://www.elastic.co/guide/en/elasticsearch/reference/current/es-release-notes.html
https://dev.mysql.com/doc/relnotes/mysql/8.0/en/

That way, you only need to display the PIM version and go on the website page. But I confess it's less convenient.

**Comment by jmleroux** [comment]
CLA check

**Comment by ahocquard** [comment]
@jmleroux I think you should talk with @NolwennP about that, because it's part of our subject also :)

**Comment by NolwennP** [comment]
I'm not sure to see which PIM persona is going to use this screen.
Also, does it actually give more value than checking the Github changelog?

**Comment by jmleroux** [comment]
> I'm not sure to see which PIM persona is going to use this screen.

Peter. I was also thinking about not providing any public link and just have an url. Because it could be also very useful for all technical teams. I know that our SREs and Support mates already love this feature. Also not providing a link and not making it public would allow us to remove/replace it when we'll have something more advanced.

> Also, does it actually give more value than checking the Github changelog?

Yes, because it merges and sort by date all the included versions.
Currently, being able to tell if a fix done in 3.2 is present in master is quite a challenge.

**Comment by jmleroux** [comment]
Anyway, the last comment was just a CLA check and I'm aware that this PR will probably not be accepted.

**Comment by ahocquard** [comment]
@jmleroux 
> Currently, being able to tell if a fix done in 3.2 is present in master is quite a challenge.

Github provides this information if you have the PR. You know which branch (including tag) has the fix... and you have the PR with Jira ticket :thinking: At least, it's how I do because it's (much) more robust than the changelog. Maybe people don't know this feature?

**Comment by jmleroux** [comment]
@ahocquard this is an answer we could give to developers used to git, but not to Peter, or partners, or even SREs who are less used to git.

**Comment by jmleroux** [comment]
I don't know if it's called a customer study, but I can tell you that every non dev person i showed the feature just wanted it. There is obviously better to do and I totally trust you to do it, but in the meantime we just have nothing. Up to you to choose between a non ideal working solution and nothing.

**Comment by jmleroux** [comment]
Anyway, this is just a PoC :wink: 
I close it to avoid misunderstanding.

**Pull request closed without merging**